src/source_sampling.h

```diff
@@ -88,24 +89,24 @@ namespace pyne {
     std::vector<double> prob; /// Probabilities.
     std::vector<int> alias; /// Alias probabilities.
   };
-  
+
   /// Problem modes
-  enum Mode {USER, ANALOG, UNIFORM};
-  
+  enum Mode {USER, ANALOG, UNIFORM, SUBVOXEL_ANALOG, SUBVOXEL_UNIFORM, SUBVOXEL_USER};
```

Comment:
The naming of this enumerator concerns me a bit. Is there a way around declaring 3 additional Modes? I'm thinking of a scheme in which you first decide if you're going to be doing subvoxel sampling and then determine if the sampling routine will be analog, uniform, or user-defined. What this might look like is a separate enumerator: 

```
enum Mode {USER, ANALOG, UNIFORM};
enum subMode{DEFAULT, SUBVOXEL};
```
Or something like that...
Then the if statement checks for the mode could go from:

```
  if (mode == USER || mode == UNIFORM || mode == ANALOG) {
    samp.push_back(-1.0);
  }
  if (mode == SUBVOXEL_USER || mode == SUBVOXEL_UNIFORM || mode == SUBVOXEL_ANALOG) {
    samp.push_back(double(cell_number[ve_idx*max_num_cells + c_idx]));
  }
```

could go to 


```
  if ( subMode == SUBVOXEL ) {
      samp.push_back(double(cell_number[ve_idx*max_num_cells + c_idx]));
    }
  else {
      samp.push_back(-1.0);
    }
```

Reply:
This would likely simplify other, similar checks in other places in the code as well.

Reply:
Thank your for the suggestion @pshriwise. I agree with you. Another variable SubMode is a good way to simplify the code and make the code easier to understand. I applied this change in the latest commit.

Reply:
Yes, the changes much cleaner to me now. I didn't put too much thought into the name of that enumerator when writing that example, so you may want to reconsider it's name. Would subvoxel sampling vs. default sampling be a "major" or "minor" mode to you? 

I guess I think of it as there now being two major modes:
  1. Subvoxel Sampling with ANALOG, UNIFORM, and USER options
and 
  2. Default sampling with ANALOG, UNIFORM and USER options

But maybe it's really the other way around. I don't know. :smile:

Reply:
I do have another option about the mode names. These modes are actually in two different level, i.e. there are two decisions to determine the mode. First, use sub-voxel r2s or not, we define this using `SubMode = {DEFAULT, SUBVOXEL}`. Second, how to bias the photon source? We define this using `BiasMode = {ANALOG, UNIFORM, USER}`. (This `BiasMode` is the original `Mode`). 

By replacing the original `Mode` with `BiasMode`, and add the parameter `SubMode`, it will be easier to understand the decision process. 

By the way, I think `SubMode = {DEFAULT, SUBVOXEL}` is a good name, easy to understand.

Reply:
Would `SampleMode` not be a little more clear if the other can be named `BiasMode`?